<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [ ]:
# Membuat dataframes dari dictionary
import pandas as pd

asia_df = pd.DataFrame({'Asia Timur': ['Jepang', 'Korea Selatan', 'Korea Utara', 'China', 'Taiwan'],
                        'Asia Tenggara': ['Indonesia', 'Malaysia', 'Singapura', 'Thailand', 'Laos'],
                        'Asia Barat': ['Arab Saudi', 'Yaman', 'Suriah', 'Yordania', 'Lebanon'],
                        'Asia Selatan': ['India', 'Nepal', 'Srilanka', 'Pakistan', 'Bangladesh']})
print(asia_df)
print("\n",type(asia_df))

      Asia Timur Asia Tenggara  Asia Barat Asia Selatan
0         Jepang     Indonesia  Arab Saudi        India
1  Korea Selatan      Malaysia       Yaman        Nepal
2    Korea Utara     Singapura      Suriah     Srilanka
3          China      Thailand    Yordania     Pakistan
4         Taiwan          Laos     Lebanon   Bangladesh

 <class 'pandas.core.frame.DataFrame'>


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [ ]:
# Membaca data dan memasukannya kedalam suatu variabel
df = pd.read_csv('diamonds-1.csv')
df

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
1. Perbedaan antara list dan series antara lain:
  - series ditampilkan (print) beserta dengan indexnya, sedangkan list hanya menampilkan value saja
  - series tidak harus mempunyai index bertipe integer atau bisa berupa sting seperti dictionary, sedangkan list hanya bisa memiliki index integer
  - series bisa melakukan slicing pada data di dalamnya seperti list, sedangkan dictionary tidak bisa langsung di-slice
  - series bisa memiliki key (index) yang duplikat, sedangkan dictionary tidak

2. Perbedaan antara dataframe dengan series adalah dataframe merupakan objek 2 dimensi dan memiliki 2 index, yaitu index baris dan index kolom. Sedangkan list hanya objek 1 dimensi dan memiliki 1 index saja. Berdasarkan perbedaan tersebut, dataframe dapat menyimpan data dengan lebih terstruktur layaknya tabel dibandingkan dengan series.

In [ ]:
# code here
series_belanja = pd.Series(["Bawang Merah","Bawang Putih","Cabe Merah","Cabe Rawit","Beras","Minyak Goreng"])
print(series_belanja)
print("\n",type(series_belanja))

0     Bawang Merah
1     Bawang Putih
2       Cabe Merah
3       Cabe Rawit
4            Beras
5    Minyak Goreng
dtype: object

 <class 'pandas.core.series.Series'>


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [ ]:
# code here
# Menampilkan  baris pertama dataframe dengan method .head() (nilai default 5 baris)
print(df.head())
# Menampilkan baris terakhir datafram dengan method .tail() (nilai default 5 baris)
print(df.tail())

# Slicing untuk kolom price dan menampilkan hanya baris paling pertama dengan method .head()
print("Baris paling pertama dari kolom price memiliki nilai", df['price'].head(1).to_string(index=False))
# Slicing untuk kolom color dan menampilkan hanya baris paling terakhir dengan method .tail()
print("Baris paling pertama dari kolom price memiliki nilai", df['color'].tail(1).to_string(index=False))

   Unnamed: 0  carat      cut color clarity  ...  table  price     x     y     z
0           1   0.23    Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43
1           2   0.21  Premium     E     SI1  ...   61.0    326  3.89  3.84  2.31
2           3   0.23     Good     E     VS1  ...   65.0    327  4.05  4.07  2.31
3           4   0.29  Premium     I     VS2  ...   58.0    334  4.20  4.23  2.63
4           5   0.31     Good     J     SI2  ...   58.0    335  4.34  4.35  2.75

[5 rows x 11 columns]
       Unnamed: 0  carat        cut color  ... price     x     y     z
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[5 rows x 11 columns]
Baris paling pertama dari kolom price memi

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: Deskriptif statistik atau statistika deskriptif adalah jenis metode statistika yang berhubungan dengan pengumpulan dan penyajian data sampai bisa memberikan informasi yang berguna. Adapun statistika deskriptif biasa digunakan untuk melihat gambaran awal mengenai data.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [ ]:
# code here
# Mengetahui deskriptif statistik dari suatu data dengan pandas
df.describe()

Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

Jawab:
1. Nilai rata-rata dari kolom price adalah :  3932.799722	
2. Nilai standar deviasi dari kolom column depth adalah :  1.432621
3. Nilai maximum dari kolom carat adalah :  5.01

In [ ]:
# Mengetahui nilai rata-rata dari kolom price
print("rata-rata dari kolom price adalah : ", df['price'].describe()['mean'])
# Mengetahui standar deviasi dari kolom depth
print("standar deviasi dari kolom column depth adalah : ", df['depth'].describe()['std'])
# Mengetahui nilai maximum dari kolom carat
print("nilai maximum dari kolom carat adalah : ", df['carat'].describe()['max'])

rata-rata dari kolom price adalah :  3932.799721913237
standar deviasi dari kolom column depth adalah :  1.4326213188337733
nilai maximum dari kolom carat adalah :  5.01
